backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxSparkPlanExecApi.scala

```diff
@@ -706,50 +706,6 @@ class VeloxSparkPlanExecApi extends SparkPlanExecApi {
     VeloxGetStructFieldTransformer(substraitExprName, childTransformer, ordinal, original)
   }
 
-  /**
-   * To align with spark in casting string type input to other types, add trim node for trimming
-   * space or whitespace. See spark's Cast.scala.
-   */
-  override def genCastWithNewChild(c: Cast): Cast = {
-    // scalastyle:off nonascii
-    // Common whitespace to be trimmed, including: ' ', '\n', '\r', '\f', etc.
-    val trimWhitespaceStr = " \t\n\u000B\u000C\u000D\u001C\u001D\u001E\u001F"
-    // Space separator.
-    val trimSpaceSepStr = "\u1680\u2008\u2009\u200A\u205F\u3000" +
-      ('\u2000' to '\u2006').toList.mkString
-    // Line separator.
-    val trimLineSepStr = "\u2028"
```

Comment:
Have you checked if they have the same sementic between vanilla Spark and velox?

Reply:
```
val expect = Seq(Row(123))
checkAnswer(sql("SELECT cast('\\t\\n\\u001F 123\\u000B' as int);"), expect)
```

this UT works.

Reply:
Maybe add a config, default false, for it? A little worried about the result diff if we haven't tested the total case between vanilla spark and velox.

Reply:
> Maybe add a config, default false, for it? A little worried about the result diff if we haven't tested the total case between vanilla spark and velox.

@Henry2SS would you?

Reply:
> > Maybe add a config, default false, for it? A little worried about the result diff if we haven't tested the total case between vanilla spark and velox.
> 
> @Henry2SS would you?

I'll do it later.

Reply:
After looking at the velox code, the conversion from string to int/long is basically similar between [vanilla Spark](https://github.com/apache/spark/blob/master/common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java#L200) and [velox](https://github.com/facebookincubator/velox/blob/main/velox/functions/lib/string/StringImpl.h#L354). But for the conversion from string to float/double, [vanilla Spark](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L390) uses `String.trim` to remove characters smaller than spaces, which is not same as velox, so we still need this extra trim.
A new config, which is default false, is good to me. Thanks @Henry2SS for working on it.

Reply:
@Henry2SS, Spark UT doesn't cover many corner cases used in production workload. I'm afraid users would complain result mismatching issues due to this removement. Agree with @jackylee-ch, it's good to use a config to give the control to users. And maybe at some future time point, we can remove it.

Reply:
> @Henry2SS, Spark UT doesn't cover many corner cases used in production workload. I'm afraid users would complain result mismatching issues due to this removement. Agree with @jackylee-ch, it's good to use a config to give the control to users. And maybe at some future time point, we can remove it.

Done.
And thanks for your advice. @PHILO-HE @jackylee-ch @Yohahaha